modules/renter/contractor/persist.go

```diff
@@ -120,6 +133,26 @@ func (c *Contractor) load() error {
 		}
 	}
 
+	// COMPATv1.1.0
+	//
+	// Wwhen loading old contracts, the corresponding relationships may not be
+	// known. Use the hostdb to make sure the relationship set is fully filled
+	// out.
+	allHosts := c.hdb.AllHosts()
+	hmap := make(map[modules.NetAddress]modules.HostDBEntry)
+	// Iterate backwards (lowest score to highest) so that in the event of
+	// duplicate hosts for a netaddress, the highest score host is the one in
+	// the map.
+	for i := len(allHosts) - 1; i >= 0; i-- {
+		hmap[allHosts[i].NetAddress] = allHosts[i]
+	}
```

Comment:
If it's not in the AllHosts definition that they will be returned in sorted order, I will add it. It's much more useful to see how the hosts stack up.